libtenzir/src/catalog.cpp

```diff
@@ -356,6 +373,13 @@ catalog_state::lookup_impl(const expression& expr, const type& schema) const {
           }();
           return result;
         },
+        [&](const data_extractor& lhs,
+            const data& d) -> catalog_lookup_result::candidate_info {
+          if (compatible(lhs.type, x.op, d)) {
+            return all_partitions();
+          }
+          return {};
+        },
```

Comment:
Afaik `data_extractor`s are only produced by the `resolve(expression, type)` function, which is not used in the catalog. I suggest to trigger an assertion if we reach this overload, then we can at least catch it in the unit tests.

Reply:
`data_extractor` is produced by `tailor`, i.e., when we bind an expression to a schema.

Reply:
I see. But then this overload effectively disables the sparse index prefiltering.

As far as I understand we should be able to fix the regression with the changes to `export` alone. Let's move the rest into another PR where we can discuss the changes more in depth.